app/lib/common/widgets/avatar/full_screen_avatar_page.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/utils/utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:zoom_hover_pinch_image/zoom_hover_pinch_image.dart';
+
+class FullScreenAvatarPage extends ConsumerWidget {
+  final String roomId;
+
+  const FullScreenAvatarPage({super.key, required this.roomId});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Scaffold(
+      appBar: _buildAppBar(context, ref),
+      body: _buildBody(context, ref),
+    );
+  }
+
+  AppBar _buildAppBar(BuildContext context, WidgetRef ref) {
+    final membership = ref.watch(roomMembershipProvider(roomId)).requireValue;
+    return AppBar(
+      leading: IconButton(
+        onPressed: context.pop,
+        icon: const Icon(Icons.close),
+      ),
+      actions: [
+        if (membership?.canString('CanUpdateAvatar') == true)
+          IconButton(
+            onPressed: () => uploadAvatar(ref, context, roomId),
+            icon: const Icon(Icons.edit_outlined),
+          ),
```

Comment:
When seeing the videos I was wondering about resetting the avatar ... like, just removing it... but we don't even have the rust API ready for that yet, so ... nothing here for now ;) .

Reply:
I also thought on the same during this PR development, but kept it for later.